### PR TITLE
[go_router] Don't clear listeners when logging is disabled

### PR DESCRIPTION
**Pull request by johnpryan**
go_router should not clear listeners on the root logger. See also https://github.com/dart-lang/logging/pull/118

cc: @devoncarew 

Fixes flutter/flutter#106653

### CODE REVIEW

File: packages/go_router/test/logging_test.dart [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-import 'package:flutter_test/flutter_test.dart';
-import 'package:go_router/src/logging.dart';
-import 'package:logging/logging.dart';
-
-void main() {
-  // Reset the logger before each test.
-  setUp(() {
```
Comment (chunhtai) [inline packages/go_router/test/logging_test.dart:11]: Instead of removing this, can you write a test to make sure setlogging does not remove listener?

**Review by chunhtai — COMMENTED** [review]
LGTM

File: packages/go_router/test/logging_test.dart (hunk 1 above)
Comment (johnpryan) [reply packages/go_router/test/logging_test.dart:11]: good idea, done.

**Comment by DartAndrik** [comment]
You guys are incredible. Thank you. Just one more thing.
Could you add `@visibleForTesting` for `setLogging` or make it private and used for GoRouter instantiation only? As `setLogging` can be called after the GoRouter instance already configured like in the example for this package:
```
 final _router = GoRouter(
    debugLogDiagnostics: true,
    routes: [
      GoRoute(
        path: '/',
        builder: (context, state) => const Page1Screen(),
      ),
      GoRoute(
        path: '/page2',
        builder: (context, state) => const Page2Screen(),
      ),
    ],
  );
```
It may be confusing why `setLogging(false)` didn't do such switching off package logging.

**Review by chunhtai — APPROVED** [review]
LGTM

**Comment by chunhtai** [comment]
@johnpryan It looks like there is merge conflict

**Pull request merged**